tools/state-viewer/src/commands.rs

```diff
@@ -1308,6 +1320,24 @@ pub(crate) fn print_state_stats(home_dir: &Path, store: Store, near_config: Near
     }
 }
 
+/// Persists the trie changes expressed by `apply_result` in the given storage.
+pub(crate) fn maybe_save_trie_changes(
+    store: Option<Store>,
+    genesis_height: u64,
+    apply_result: ApplyChunkResult,
+    block_height: u64,
+    shard_id: u64,
+) -> anyhow::Result<()> {
+    if let Some(store) = store {
+        let mut chain_store = ChainStore::new(store, genesis_height, true);
+        let mut chain_store_update = chain_store.store_update();
+        chain_store_update.save_trie_changes(apply_result.trie_changes);
+        chain_store_update.commit()?;
```

Comment:
Ah I think I know how does this write into State: 
https://github.com/near/nearcore/blob/c422664bf91f75d6d29ffb19d2b16df5d71c263a/chain/chain/src/store/mod.rs#L2539-L2543

The insertions_into may also write to State but please confirm this. I'm worried if any of the other "side effects" of writing trie changes may hurt us - e.g. 
* storing the trie nodes with arbitrary ref count - The cold storage may have some assertions on it 
* storing deletions - Can that potentially delete some trie nodes from the storage? 
* storing state changes - Those are not typically stored in cold storage
* storing the trie changes itself - Those are not typically stored in cold storage

I think we may want to go one level lower and work directly on the State column. We may also want to focus just on the cold storage and reuse some of the cold storage methods to make sure we store the data in the right format. 

btw I love your approach and implementation it's really clean!

Reply:
Yeah ignore my message below it's saying the same thing :)

I'm preoccupied by somewhat losing data (GC or anything else). Having the node running for 5+ epochs and restarting should give confidence.

Yes deletions can happen in theory - the same deletions that happened naturally at the time I think. Also the changes are applied only if chunk extra matches so we should be pretty safe.

About the cold storage, yes we should verify data is stored correctly! Need to find how

Reply:
Thanks for introducing the recovery db, it handles limiting ops to State and writing (no-deletions) really neatly! 

Did you get a chance to make sure that all the ops go through the colddb's `adjust_op` to ensure the format is correct?